dota2/features/lobby.py

```diff
@@ -0,0 +1,348 @@
+
+    def request_friend_practice_lobby_list(self):
+        """
+        Request a list of friend practice lobbies.
+        """
+        if self.verbose_debug:
+            self._LOG.debug("Requesting friend practice lobby list.")
+
+        jobid = self.send_job(EDOTAGCMsg.EMsgGCFriendPracticeLobbyListRequest, {})
+
+        @self.once(jobid)
+        def wrap_lobby_list(message):
+            self.emit(self.EVENT_LOBBY_FRIEND_LIST_ANSWER, message)
+
+        return jobid
+
+    def balanced_shuffle_lobby(self):
+        """
+        Balance shuffle the the lobby.
+        """
+        if self.verbose_debug:
+            self._LOG.debug("Balance shuffle the the lobby")
+
+        self.send(EDOTAGCMsg.EMsgGCBalancedShuffleLobby, {})
+
+    def flip_lobby_teams(self):
+        """
+        Flip both teams of the lobby.
+        """
+        if self.verbose_debug:
+            self._LOG.debug("Flipping teams of the lobby")
+
+        self.send(EDOTAGCMsg.EMsgGCFlipLobbyTeams, {})
+
+    def invite_to_lobby(self, steam_id):
+        """
+        Asks to invite a player to your lobby. This creates a new default lobby when you are not already in one.
+
+        :param steam_id: steam_id
+        :type steam_id: :class:`int`
+        """
+        if self.verbose_debug:
+            self._LOG.debug("Inviting %s to a lobby." % steam_id)
+
+        self.send(EGCBaseMsg.EMsgGCInviteToLobby, {
+            "steam_id": steam_id
+        })
+
+    def practice_lobby_kick(self, account_id):
+        """
+        Kick a player from the lobby.
+
+        :param account_id: steam_id
+        :type account_id: :class:`int`
+        """
+        if self.verbose_debug:
+            self._LOG.debug("Kicking %s from the lobby." % account_id)
+
+        self.send(EDOTAGCMsg.EMsgGCPracticeLobbyKick, {
+            "account_id": account_id
+        })
+
+    def practice_lobby_kick_from_team(self, account_id):
+        """
+        Kick a player from the his current lobby team.
+
+        :param account_id: account_id
+        :type account_id: :class:`int`
+        """
+        if self.verbose_debug:
+            self._LOG.debug("Kicking %s from his lobby team." % account_id)
+
+        self.send(EDOTAGCMsg.EMsgGCPracticeLobbyKickFromTeam, {
+            "account_id": account_id
+        })
+
+    def join_practice_lobby(self, id, password=""):
+        """
+        Join the target practice lobby.
+
+        :param id: id of the lobby to join
+        :type password: :class:`str` pass phrase of the lobby to join
+        """
+        if self.verbose_debug:
+            self._LOG.debug("Trying to join practice lobby %s using password %s" % (id, password))
+
+        jobid = self.send_job(EDOTAGCMsg.EMsgGCPracticeLobbyJoin, {
+            "lobby_id": id,
+            "pass_key": password
+        })
+
+        @self.once(jobid)
+        def wrap_join_lobby(message):
+            self.emit(self.EVENT_LOBBY_JOIN_ANSWER, message)
+
+        return jobid
```

Comment:
I don't think this is necessary to emit an event. You can just wait for the response and return the result.

```python
jobid = self.send_job(EDOTAGCMsg.EMsgGCPracticeLobbyJoin, {
                       "lobby_id": id,
                       "pass_key": password
                       })

resp = self.wait_msg(jobid, timeout=10)

return DOTAJoinLobbyResult(resp.result) if resp else DOTAJoinLobbyResult.DOTA_JOIN_RESULT_TIMEOUT